src/refdb_fs.c

```diff
@@ -962,6 +962,8 @@ static int packed_write(refdb_fs_backend *backend)
 
 	for (i = 0; i < git_sortedcache_entrycount(refcache); ++i) {
 		struct packref *ref = git_sortedcache_entry(refcache, i);
+		if (!ref)
+			goto fail;
```

Comment:
Ah, okay. I've seen the other loops where we iterate over the cache using the same pattern. Didn't realize we acquire a write lock here, so `assert`ing this should indeed be the way to go.